### ISSUE TITLE
`AssertionError` in `invalidation_trees`: `ncovered > 0`

### ISSUE THREAD

**Issue by jishnub** [timholy/SnoopCompile.jl#323]
This is using v1.5.0 of `FFTW.jl`
```julia
julia> using SnoopCompileCore

julia> invalidations = @snoopr using FFTW;

julia> using SnoopCompile

julia> trees = invalidation_trees(invalidations)
ERROR: AssertionError: ncovered > 0
Stacktrace:
 [1] (::SnoopCompile.var"#handle_insert_backedges#138")(list::Vector{Any}, i::Int64, callee::Core.MethodInstance)
   @ SnoopCompile ~/.julia/packages/SnoopCompile/Q8zUg/src/invalidations.jl:321
 [2] invalidation_trees(list::Vector{Any}; exclude_corecompiler::Bool)
   @ SnoopCompile ~/.julia/packages/SnoopCompile/Q8zUg/src/invalidations.jl:387
 [3] invalidation_trees(list::Vector{Any})
   @ SnoopCompile ~/.julia/packages/SnoopCompile/Q8zUg/src/invalidations.jl:294
 [4] top-level scope
   @ REPL[4]:1

(FFTW) pkg> st
Project FFTW v1.5.0
Status `~/Dropbox/JuliaPackages/FFTW.jl/Project.toml`
  [621f4979] AbstractFFTs v1.2.1
  [21216c6a] Preferences v1.3.0
  [189a3867] Reexport v1.2.2
  [f5851436] FFTW_jll v3.3.10+0
  [856f044c] MKL_jll v2022.2.0+0
  [37e2e46d] LinearAlgebra

(FFTW) pkg> activate 
  Activating project at `~/.julia/environments/v1.8`

(@v1.8) pkg> st SnoopCompile
Status `~/.julia/environments/v1.8/Project.toml`
  [aa65fe97] SnoopCompile v2.9.5

(@v1.8) pkg> st SnoopCompileCore
Status `~/.julia/environments/v1.8/Project.toml`
  [e2b509da] SnoopCompileCore v2.9.0

julia> versioninfo()
Julia Version 1.8.4
Commit 00177ebc4fc (2022-12-23 21:32 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 8 × 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, tigerlake)
  Threads: 1 on 8 virtual cores
Environment:
  LD_LIBRARY_PATH = :/usr/lib/x86_64-linux-gnu/gtk-3.0/modules
  JULIA_EDITOR = subl
```

**Comment by sloede** [comment]
We get a similar error for the invalidations GH action for our package, e.g., [here](https://github.com/trixi-framework/Trixi.jl/actions/runs/3792502558/jobs/6448746153#step:5:235).

**Comment by dpinol** [comment]
For the record, I can reproduce it with julia 1.8.4, but not with julia 1.8.3.
And it looks like it's solved with SnoopCompile 2.9.6 :-)

**Comment by jishnub** [comment]
Yes, it does seems fixed
